csharp/device/Microsoft.Azure.Devices.Client/Message.cs

```diff
@@ -144,7 +178,11 @@ public string To
         {
             get
             {
+#if NETMF
+                return (string)(this.GetSystemProperty(MessageSystemPropertyNames.To) ?? string.Empty);
```

Comment:
I think this is still going to throw instead of returning null if the cast fails, I don't believe it's the desired effect. I'll change these (this one and the following) to "as" casts. thoughts?